### ISSUE TITLE
Question: `adamw` with `add_decayed_weights`

### ISSUE THREAD

**Issue by jenkspt** [deepmind/optax#494]
I'm trying to apply L2 penalty to specific parameters like so:

    optimizer = optax.chain(
        optax.clip_by_global_norm(grad_clip),
        # Apply weight decay excluding bias and embedding parameters
        optax.add_decayed_weights(weight_decay, mask=param_decay_mask(params)),
        optax.adamw(learning_rate, weight_decay=0.0),
    )

does add_decayed_weights + adamw perform the correctly adjusted parameter update?

**Comment by jenkspt** [comment]
Missed that the `mask` parameter for `adamw` is for masking the weight decay.
